KE-Apache-Beam-Analysis/README.md

```diff
@@ -0,0 +1,82 @@
+
+# Data Analytics @ Edge
+## Description
+![High level architecture](Images/High_level_Arch.png "High Level Architecture")
+
+The main aim of analytics engine is to get data from mqtt broker in stream format and apply rules on incoming data in real time and produce alert/action on mqtt broker. Getting data through pipeline and aplying analysis function is done by using apache beam.
+
+###  Apache beam:
+Apache Beam is an open source, unified model for defining both batch and streaming data-parallel processing pipelines. Using one of the open source Beam SDKs, wecan build a program that defines the pipeline.
+
+
+#### Why use Apache Beam for analytics:
+There are many frameworks like Hadoop, Spark, Flink, Google Cloud Dataflow, etc that came into existence. But there has been no unified API that binds all these frameworks and data sources, and provide an abstraction to the application logic from big data ecosystem. Apache Beam framework provides abstraction between your application logic and big data ecosystem. 

```

Comment:
There are many frameworks like Hadoop, Spark, Flink, Google Cloud Dataflow, etc **that came into existence.**  -->
There are many frameworks like Hadoop, Spark, Flink, Google Cloud Dataflow, etc **for stream processing**.

 But there has been no unified API that binds all these frameworks and data sources, and provide an abstraction to the application logic from big data ecosystem.  -->

There was no unified API to bind all such frameworks and data sources . It was needed to abstract out the application logic from these Big Data frameworks.

provides abstraction  --> provides this abstraction.